### DISCUSSION TITLE
Audio lags when a video is played

### DISCUSSION THREAD

**Discussion by advaithm582** [neutrinolabs/pulseaudio-module-xrdp#123] [Q&A] [answered]
The README suggested two ways of checking if the setup worked:
1. try `paplay`
2. play a youtube video in Firefox (FF). 

The first went smoothly, however, if I attempt to play any video in FF, the audio is so laggy that it is barely comprehensible. The remote PC has decent specs and can render videos perfectly in the console session. The client PC also has good specs, and are linked by a decently fast network connection.

Is there any configuration that will aid in improving the audio output while video is rendering?

For reference, I am using Windows' `mstsc.exe` as the client and xrdp 0.9.21.1 on Debian 12.

**Comment by advaithm582** [comment]
Any updates on the same? 

For more specifics, the client and the server are linked by a 1Gbps Ethernet connection. The remote PC has 16G of RAM.
Some things I tried were changing the audio compression settings in the .rdp file - these seemed to have minimal effect.

Also, I noticed that the issue mainly happens on full screen video (if it is only a portion of the screen, there is no lag.)

**Comment by matt335672** [comment, marked as answer]
There are no specific configs related to priorities of audio/video.

However, on v0.9.x you may be suffering from a TCP buffering problem. See [this post](https://github.com/neutrinolabs/xrdp/issues/2905#issuecomment-1880694050) for a config you can try.

**Reply by advaithm582** [reply]
Thanks for the response! I will try out the linked solution when I get the time.
